### ISSUE TITLE
x/build/dashboard: remove MallocNanoZone=0 from macOS 12 builder configuration

### ISSUE THREAD

**Issue by dmitshur** [golang/go#49940]
The macOS 12 builder environments had `MallocNanoZone=0` added to work around #49138. This is the tracking issue to remove that workaround once it is no longer needed.

@cherrymui [CL 367916](https://go.dev/cl/367916) has recently closed #49138; do you know if now a good time to stop using `MallocNanoZone=0`?

**Comment by cherrymui** [comment]
I think it should be okay to remove the environment variable for tip (and Go 1.18). But we still need it for 1.17. I'll work on a workaround for 1.17 then we can remove it.

**Comment by dmitshur** [comment]
Thanks Cherry. We'll wait for the 1.17 workaround to land. I think it can be removed from builder configuration when 1.18 is released and 1.16 stops being supported. So waiting on that.
